Library/Homebrew/cmd/update.sh

```diff
@@ -312,6 +318,20 @@ EOS
     QUIET_ARGS=()
   fi
 
+  if [[ -n "$HOMEBREW_DEBUG" || -n "$HOMEBREW_VERBOSE" ]]
+  then
+    if [[ -z "$HOMEBREW_UPDATE_NO_PARALLEL" ]]
+    then
+      opoo <<EOS
+Enabling debug and/or verbose mode can lead to overlapping output while
+updates are fetched in parallel. If this output is important to you, retry with:
+  brew update --no-parallel $*
+
+Note that disabling parallel fetch tends to significantly slow down the update.
+EOS
+    fi
+  fi
```

Comment:
As it creates another option (and possible edge cases) that we'll have to support.